src/MICore/LaunchOptions.cs

```diff
@@ -1792,6 +1792,27 @@ protected void InitializeCommonOptions(Json.LaunchOptions.BaseOptions options)
             this.WaitDynamicLibLoad = false;
 
             this.SourceMap = SourceMapEntry.CreateCollection(options.SourceFileMap);
+
+            if (options.SymbolLoadInfo != null)
+            {
+                this.WaitDynamicLibLoad = !String.IsNullOrWhiteSpace(options.SymbolLoadInfo.ExceptionList) || options.SymbolLoadInfo.LoadAll.HasValue;
+
+                SymbolInfoLoadAll = options.SymbolLoadInfo.LoadAll.GetValueOrDefault(true);
+
+                if (DebuggerMIMode == MIMode.Lldb && !string.IsNullOrWhiteSpace(options.SymbolLoadInfo.ExceptionList))
```

Comment:
This code feels like it would be a lot cleaner as:

```
if (!string.IsNullOrWhiteSpace(options.SymbolLoadInfo.ExceptionList))
{
    if (DebuggerMIMode == MIMode.Lldb)
        throw...

    this.WaitDynamicLibLoad = true;
    SymbolInfoExceptionList.SetTo(options.SymbolLoadInfo.ExceptionList.Split(';'));
}
else
{
    SymbolInfoExceptionList.SetTo(new string[0]);
}
```